### ISSUE TITLE
No sound on some devices

### ISSUE THREAD

**Issue by fpena** [nicklockwood/SoundManager#5]
Hi Nick,

First of all, thanks for the library.

I've developed an app with it, but for some reason, on some devices, it doesn't generate sound unless I plug the headphones. 

Have you heard this issue before?

Thanks again.

**Comment by nicklockwood** [comment]
No - which specific devices are you seeing this on?

It may be that you need to change the default audio route. I'll see if there's a simple way to handle that within the library.

**Comment by fpena** [comment]
It is very rare and I haven't experienced that personally. It seems some
people can't hear external sound but when plugging headphones, everything
is ok.

Looking forward if you find a way to change the default audio route.

Thanks.

On Tue, Aug 27, 2013 at 6:00 PM, Nick Lockwood notifications@github.comwrote:

> No - which specific devices are you seeing this on?
> 
> It may be that you need to change the default audio route. I'll see if
> there's a simple way to handle that within the library.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/nicklockwood/SoundManager/issues/5#issuecomment-23374654
> .

## 

Q: Why is this email three sentences or less?
A: http://three.sentenc.es

---

Felipe Peña Riveros
http://felipe.penya.cl

**Comment by sgabello** [comment]
@fpena I had many people with the same issue... 90% of the time (with the iPad in particular) they didn't realise that they had the mute switch enabled or they didn't know what was it for... unbelievable! :)
Some then argued that the audio was ok in other apps. That happens because there are ways to play sounds and media files that ignore the mute setting.

**Comment by fpena** [comment]
Yes. That's exactly what happened. It's ok. I'm working with MIDI so only
need the library for specific sounds. Thanks Andrea!

On Tue, Sep 17, 2013 at 6:08 AM, Andrea Ottolina
notifications@github.comwrote:

> @fpena https://github.com/fpena I had many people with the same
> issue... 90% of the time (with the iPad in particular) they didn't realise
> that they had the mute switch enabled or they didn't know what was it
> for... unbelievable! :)
> Some then argued that the audio was ok in other apps. That happens because
> there are ways to play sounds and media files that ignore the mute setting.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/nicklockwood/SoundManager/issues/5#issuecomment-24573795
> .

## 

Q: Why is this email three sentences or less?
A: http://three.sentenc.es

---

Felipe Peña Riveros
http://felipe.penya.cl

**Comment by nobre84** [comment]
Hi! I had this happen very often (maybe after iOS 8.1 ? It didn't happen before Xcode 6 for sure)
The sounds won't play in the speaker, unless I plug the earphone, and THEN if I take it out, the speakers will play audio.
My usage is as follows:

<pre>
[SoundManager sharedManager].allowsBackgroundMusic = YES;
[[SoundManager sharedManager] prepareToPlay];
[[SoundManager sharedManager] playSound:fileName looping:NO]
</pre>
